compute_tools/src/migrations/0012-grant_usage_on_foreign_data_wrapper_to_neon_superuser.sql

```diff
@@ -0,0 +1 @@
+GRANT USAGE ON FOREIGN DATA WRAPPER postgres_fdw TO neon_superuser;
```

Comment:
Yeah, migration won't work in this case. What was the actual error when you tried to use fdw by non-superuser?

> I think we need to add an extension hook within Neon that just runs SQL when an extension is installed.

This sounds a bit too complicated, but it may work. I wonder, can we instead add some permission validation hook that will treat `neon_superuser` as superuser in certain cases? It might be hard, because we don't want to put it into `superuser()` yet, it will make it too broad and probably real superuser. But maybe there are other places we can interfere to only escalate permissions in specific cases

Reply:
"permission denied for foreign-data wrapper postgres_fdw"

Reply:
There seem to be a common entry point for all ACL checks if user is not an owner of the object
https://github.com/neondatabase/postgres/blob/a8dd6e779dde907778006adb436b557ad652fb97/src/backend/catalog/aclchk.c#L3112

We can add a hook there

Then it's a question of what we will do in this hook. I guess OIDs could be arbitrary there, but maybe we can extract an object name as well and build some allow list?

I'd check the code around and start a #postgresql thread because I'm not sure that what I propose is a good idea